doc/samtools-mpileup.1

```diff
@@ -482,7 +483,43 @@ The
 .B -E
 option can be used to make it ignore the contents of the BQ:Z tag and
 force it to recalculate the BAQ scores by making a new alignment.
-
+.PP
+.SH EXAMPLES
+.PP
+Using range:
+.PP
+with implicit index files in1.bam.<ext> and in2.sam.gz.<ext>,
+.IP
+.B samtools mpileup in1.bam in2.sam.gz -r <name>:<range>
+.PP
+with explicit index files,
+.IP
+.B samtools mpileup in1.bam in2.sam.gz <in1 index path> <in2 index path> -X -r <name>:<range>
+.PP
+with fofn having input file names, with implicit index files present along with inputs,
+.IP
+.B samtools mpileup -b fofn -r <name>:<range>
+.PP
+Using flags:
+.PP
+to get reads with flags READ2 or REVERSE and not having any of SECONDARY,QCFAIL,DUP,
+.IP
+.B samtools mpileup --rf READ2,REVERSE in.sam
+.R or
+.B samtools mpileup --rf 144 in.sam
+.PP
+to get reads with flag SECONDARY,
+.IP
+.B samtoos mpileup --rf SECONDARY --ff QCFAIL,DUP in.sam
+.R or
+.B samtools mpileup --rf 256 --ff 1536 in.sam
+.PP
+Using all possible alignmentes:
+.PP
+i.e. counting orphans, avoiding BAQ calculation, using max depth, min BAQ as 0, exclude flag as 0 and overlap removal disabled
```

Comment:
BAQ is somewhat orthogonal to showing all alignments, but I think it is a reasonable thing to include as typically you're wanting all data because you'd like it in the rawest possible form, which also means not munging qualities.

Note though `-Q 0` is "using a minimum base quality of 0", not "min BAQ as 0".  That's a small distinction, but we've just stated with `-B` that we're avoiding BAQ so people may think the `-Q` then has no effect when phrased as-is.

Also as this is a proper man page we ought to use "minimum" and "maximum" instead of "min" and "max".

Similarly "exclude flag as 0" could be "exclude flag as 0 (none excluded)".
Finally, given this is a long list of things, maybe the Oxford comma here helps clarify, so ", and overlap removal disabled".

The alternative though to describing the command line and then showing the command line, would be to just use the long-opts, maybe with the short one underneath it.  Eg:

```
To show all possibile alignments, these two equivalent commands may be used.
These also disables BAQ so the quality values are presented in their original form.
.EX 2
samtools mpileup --count-orphans --no-BAQ --max-depth 0 --excl-flags 0 \\
    --disable-overlap-removal --min-BQ 0 in.bam

samtools mpileup -A -B -d 0 --ff 0 -x -Q 0 in.bam
.EE
```